synapse/storage/databases/main/event_federation.py

```diff
@@ -93,13 +89,12 @@ def _get_auth_chain_ids_txn(self, txn, event_ids, include_given, ignore_events):
                 txn.execute(base_sql + clause, args)
                 new_front.update(r[0] for r in txn)
 
-            new_front -= ignore_events
             new_front -= results
 
             front = new_front
             results.update(front)
 
-        return list(results)
+        return results
```

Comment:
most of the callers can cope with a set here, so we may as well return it rather than converting to a list.

Reply:
Unless I'm missing something this seems to only get called in two places and both of them will want lists?
In `get_auth_chain` above and in the `FederationServer._on_state_ids_request_compute` -- I think the second one gets used as a response and I don't think sets are JSON serializable.

Reply:
Looks like the sytests failed with this error.

Reply:
oh feh. this PR is rapidly turning into a waste of time.